plugin html edit jodit

minhnhut16 opened the pull request:
## Description

Currently, I'm going to create a custom plugin for HTML editor with [jodit](https://xdsoft.net/jodit/) because default HTML editor (quill) cannot adapt to my business. 
I'm following this guide [https://www.builder.io/c/docs/make-a-plugin](https://www.builder.io/c/docs/make-a-plugin)

I'm using node 16.18.0

_Screenshot_
<img width="891" alt="Screenshot 2024-05-24 at 16 33 47" src="https://github.com/BuilderIO/builder/assets/89378014/5893314a-2f45-41ae-84f7-ab287271ba34">

minhnhut16 commented:
@gustavocodess Could you approve this PR for me ? Is it have any issue?

teleaziz commented on plugins/html-editor/README.md:
```diff
@@ -0,0 +1,89 @@
+# Builder.io example plugin
```
readme needs a few instructions on how to enable and what this plugin do, and how to use it

teleaziz commented on plugins/html-editor/src/plugin.tsx:
```diff
@@ -0,0 +1,36 @@
+  /**
+   * Here we override the built-in richtext editor.
+   */
+  name: 'html-editor',
```
this should be the name of a type, if you want to override built-in rich text editor you'd want to chose `html` or `richText`

teleaziz reviewed (changes requested):
Great suggestion, left few comments otherwise looks good

minhnhut16 commented:
hi BuilderIO team, I have updated on pull request, please check and give me
feedback if have any update

On Fri, Jul 12, 2024 at 12:29 AM Aziz Abbas ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Great suggestion, left few comments otherwise looks good
> ------------------------------
>
> In plugins/html-editor/README.md
> <https://github.com/BuilderIO/builder/pull/3306#discussion_r1674393444>:
>
> > @@ -0,0 +1,89 @@
> +# Builder.io example plugin
>
> readme needs a few instructions on how to enable and what this plugin do,
> and how to use it
> ------------------------------
>
> In plugins/html-editor/src/plugin.tsx
> <https://github.com/BuilderIO/builder/pull/3306#discussion_r1674400023>:
>
> > +
> +function RichTextEditor(props: TextProps) {
> +  return (
> +    <JoditEditor
> +      value={props.value}
> +      onChange={props.onChange}
> +      config={config}
> +    />
> +  );
> +}
> +
> +Builder.registerEditor({
> +  /**
> +   * Here we override the built-in richtext editor.
> +   */
> +  name: 'html-editor',
>
> this should be the name of a type, if you want to override built-in rich
> text editor you'd want to chose html or richText
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BuilderIO/builder/pull/3306#pullrequestreview-2172538515>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVJ4ZXWABI7A27MXMTPQXMLZL26GLAVCNFSM6AAAAABIKUQAPSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCNZSGUZTQNJRGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
CONFIDENTIALITY NOTE:
This email (including any attachments) contains 
confidential information and is intended only for the recipient(s) 
addressed above. Please notify the sender immediately if you have received 
this e-mail in error and destroy it (and all attachments) without reading, 
storing or disseminating any of its contents (in any form) to any person. 
Email communication is not secure. Trusting Social Group are not liable for 
any losses arising out of any errors or omissions in the contents resulting 
from email transmission or any illegal or unauthorized usage or tampering 
of our email system.

teleaziz commented on plugins/html-editor/README.md:
````diff
@@ -0,0 +1,24 @@
+# Jodit plugin to override built-in rich text
+
+The built-in rich text editor (Quill) cannot meet the needs of my business. Currently, I am using another rich text editor, [Jodit](https://xdsoft.net/jodit), which includes additional features to serve the business such as tables, videos, symbols, and more.
+
+### How to run this plugin
+Video demo on local [here](https://drive.google.com/file/d/1b_f4hLewbHUH_KfxxijWzuVEmNas29xk/view) 
+
+using node version 16.18.0
+
+```bash
+git clone https://github.com/BuilderIO/builder.git
+cd plugins/html-editor
+npm install
+```
+
+Test plugin on local
+```bash
+npm run start
+```
+
+Build js file on production
+```bash
+npm run build
+```
````
README needs significant improvement before we can consider adding this plugin. Here are some suggestions:

**Introduction**: Please provide a clear, concise introduction that explains what the plugin does and its benefits. The current description is too specific to your use case.

**Installation**: Include clear instructions on how to install the plugin in our app, not just how to run it locally, this section could help https://www.builder.io/c/docs/make-a-plugin#step-4-add-your-plugin-to-builder , except instead of localhost user would type the package name from this plugin `@builder.io/plugin-html-editor`

**Remove personal references**: The README should be generic and not reference your specific business needs.

**Video demo**: Instead of a local file link, consider hosting the demo video on a public platform for easier access, or making a gif, it's okay to omit and use screenshots as well.

Please revise the README addressing these points. This will make it easier for us to evaluate the plugin and for other developers to understand and use

minhnhut16 replied:
@teleaziz  `except instead of localhost user would type the package name from this plugin @builder.io/plugin-html-editor`

I don't understand what you mean. Here, I can only run it locally. You need to publish my plugin to your NPM registry for me to be able to do this, right?

Or, if I try to host this plugin on my personal NPM registry, it will give this error:
<img width="727" alt="Screenshot 2024-07-15 at 16 34 57" src="https://github.com/user-attachments/assets/f059b0b6-7942-473d-989b-e67f2a0aaa78">

minhnhut16 replied:
hello @teleaziz

minhnhut16 replied:
I have updated README following your comment @teleaziz
